modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/endpointComponents/PathParameterConfiguration.tsx

```diff
@@ -0,0 +1,209 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert from '@clayui/alert';
+import {Text} from '@clayui/core';
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import classNames from 'classnames';
+import {sub} from 'frontend-js-web';
+import React, {
+	Dispatch,
+	SetStateAction,
+	useCallback,
+	useEffect,
+	useState,
+} from 'react';
+
+import {Select} from '../fieldComponents/Select';
+import {DEFAULT_LANGUAGE_ID} from '../utils/constants';
+import {fetchJSON, getAllItems} from '../utils/fetchUtil';
+import {removeLeadingForwardSlash} from '../utils/string';
+
+interface PathParameterConfigurationProps {
+	data: Partial<APIEndpointUIData>;
+	displayError: EndpointDataError;
+	selectedResponseBodySchema: SelectOption | undefined;
+	setData: Dispatch<SetStateAction<Partial<APIEndpointUIData>>>;
+}
+
+export default function PathParameterConfiguration({
+	data,
+	displayError,
+	selectedResponseBodySchema,
+	setData,
+}: PathParameterConfigurationProps) {
+	const [pathParameterOptions, setPathParameterOptions] = useState<
+		SelectOption[]
+	>([]);
+
+	const [selectedPathParameter, setSelectedPathParameter] = useState<
+		SelectOption
+	>();
+
+	useEffect(() => {
+		setSelectedPathParameter(
+			pathParameterOptions.find(
+				(option) => option.value === data.pathParameter
+			)
+		);
+	}, [pathParameterOptions, data.pathParameter]);
+
+	const getObjectFieldLabel = (objectField: ObjectDefinition): string => {
+		if (
```

Comment:
This functions seems complex, even to understand what is it doing. Is there any way to simplify it? 

I now this is would probably won't work, but something aproximate would be easier to understand. 

`return objectField.label[DEFAULT_LANGUAGE_ID as keyof LocalizedValue<string>] || Object.values(objectField.label)?[0] || objectField.name;`